docs/how-to/environment/use-environments.md

```diff
@@ -0,0 +1,131 @@
+
+```toml config-example
+[tool.hatch.envs.default]
+dependencies = [
+  "pydantic",
+  "numpy",
+]
+```
+
+You can declare versions for your dependencies as well within this configuration. 
+
+```toml config-example
+[tool.hatch.envs.default]
+dependencies = [
+  "pydantic>=2.0",
+  "numpy",
+]
+```
+
+### Create custom development environment  
+
+You can create custom environments in Hatch by adding a section to your pyproject.toml file `[tool.hatch.envs.environment-name]`. Below you define an environment called test and you add the pytest dependency to that environment's configuration. 
+
+```toml config-example
+[tool.hatch.envs.test]
+dependencies = [
+  "pytest"
+]
+```
+
+The first time that you call the test environment, hatch will:
+
+1. create the environment
+2. install all dependencies defined in your pyproject.toml file
+3. install your project in editable mode 
+
+### Run commands within a specific environment
+
+Hatch offers a unique environment feature 
+that allows you run a specific command within a specific environment rather than needing to activate the environment as you would using a tool such as conda or venv. 
+
+For instance, if you define an environment called tests that contains the pytest package, you can run
+the pytest command from the tests environment using the syntax:
+
+`hatch -e environment-name:command`
+
+To access the test environment and run pytest, you can run:
+
+```bash
+> hatch run test:pytest
+
+============================== test session starts ===============================
+platform darwin -- Python 3.12.1, pytest-7.4.4, pluggy-1.3.0
+rootdir: /your/path/here/pyosPackage
+collected 0 items   
+```  
+
+Above:
+
+* `test:pytest` represents the name of the environment to call (`test`) and the command to run (`pytest`).
+
+## View current environments
+
+Above you defined and created a new test environment in your pyproject.toml file. You can now 
+use hatch env show to see both the currently created environments and the dependencies in each environment. 
+
+```bach
+> hatch env show
+```
+
+❯ hatch env show
+             Standalone             
+┏━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━┓
+┃ Name    ┃ Type    ┃ Dependencies ┃
+┡━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━┩
+│ default │ virtual │              │
+├─────────┼─────────┼──────────────┤
+│ test    │ virtual │ pytest       │
+└─────────┴─────────┴──────────────┘
+
+To see where your current environment is located you can use `hatch env find`.
+
+```bash
+> hatch env find
+/your/path/Application Support/hatch/env/virtual/pyospackage/twO2iQR3/pyospackage
+```
+
+## Launching an environment specific shell
+
+If you with to enter the environment that you have created rather than calling a command to run within that environment you can use 
+
+```
+> hatch shell
+```
+
+todo: launch into a shell for the test envt? how?
```

Comment:
@ofek or anyone - i can't figure out how to launch into a shell for a specific environment such as test. i tried

hatch shell test 

hatch run test:shell (doesn't make sense but tried it)

hatch -e test shell

how do i get a user into their test shell in case they want to do some work at the CLI in that envt?

thank you!

Reply:
The one you mentioned works `hatch -e test shell`

Reply:
it does. @ofek i'm going to work on it a bit more tonight. i must have been tired last night `hatch -e test shell` works.